src/hotspot/share/opto/mulnode.cpp

```diff
@@ -63,9 +63,19 @@ Node *MulNode::Ideal(PhaseGVN *phase, bool can_reshape) {
   const Type *t2 = phase->type( in(2) );
   Node *progress = NULL;        // Progress flag
 
-  // convert "max(a,b) * min(a,b)" into "a*b".
+  // convert "(-a)*(-b)" into "a*b"
   Node *in1 = in(1);
   Node *in2 = in(2);
+  if (in1->is_Sub() && in2->is_Sub()) {
+    Node* n11 = in1->in(1);
+    Node* n21 = in2->in(1);
+    if (phase->type(n11)->is_zero_type() &&
+        phase->type(n21)->is_zero_type()) {
+      return make(in1->in(2), in2->in(2));
```

Comment:
Why do you need to create a new node? Can't you simply update the inputs like the code below does?

Reply:
Sorry, I missed your early comment. Fixed.